### PR TITLE
Laboratory work #2, Anna Egorova - 21FPL1

### PR DESCRIPTION
**Pull request by chezar0**

### CODE REVIEW

File: lab_1_keywords_tfidf/main.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def clean_and_tokenize(text: str) -> Optional[list[str]]:
     Returns:
     list[str]: A sequence of lowercase tokens with no punctuation
 
-    In case of corrupt input arguments, None is returned
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:71]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Hello, @chezar0 ! As a reminder, the deadline for the second lab is October 21th and you have less than 2 weeks to complete it. Please hurry.

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Hey @chezar0 ! As a reminder, the deadline for the second lab is October 21st and you have less than a week to complete it. Please hurry

File: lab_2_keywords_cooccurrence/main.py [hunk 2]
```diff
@@ -123,7 +200,27 @@ def extract_candidate_keyword_phrases_with_adjoining(candidate_keyword_phrases:
...
+    new_candidates = []
+    possible_candidates = []
+    new_candidates.append(pairwise(candidate_keyword_phrases))
+    no_duplicates = list(dict.fromkeys(new_candidates))
+    for i in new_candidates:
+        if i not in no_duplicates:
+            possible_candidates.append(i)
+    possible_candidates = list(dict.fromkeys(possible_candidates))
+    for i in phrases:
+        i = i.split()
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:215]: looks like this code has not effect

File: lab_2_keywords_cooccurrence/main.py [hunk 3]
```diff
@@ -123,7 +200,27 @@ def extract_candidate_keyword_phrases_with_adjoining(candidate_keyword_phrases:
...
+    phrases_with_ajoin = []
+    for i in phrases:
+        for candidate in possible_candidates:
+            phrase = r'(\b\w*\b)(?<= ' + candidate[0][-1] + ')(?= ' + candidate[-1][0] + ')'
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:219]: use f string

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_keywords_cooccurrence/main.py (hunk 2 above)
Comment (chezar0) [reply lab_2_keywords_cooccurrence/main.py]: fixed

**Pull request closed without merging**